[BUILD] build fails for target linux arm64 musl

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Building as a dependency of a program using `krm-kcl`, with `CGO_ENABLED=1` and `CC="zig cc -target aarch64-linux-musl -lunwind"` and `-tags musl`, I have the following errors:

```console
    error=
    │ build failed: exit status 1: # kcl-lang.io/lib/go/native
    │ ../go/pkg/mod/kcl-lang.io/lib@v0.12.3/go/native/native_musl_arm64.go:52:19: cannot convert (_Cfunc_kcl_service_call_with_length)(cSvc, cMethod, cArgs, cArgsLen, cOutSize) (value of type *_Ctype_uint8_t) to type uintptr
    │ ../go/pkg/mod/kcl-lang.io/lib@v0.12.3/go/native/native_musl_arm64.go:55:15: cannot use _Ctype_uintptr_t(ptr) (value of uint64 type _Ctype_uintptr_t) as *_Ctype_uint8_t value in argument to (_Cfunc_kcl_free)
    target=linux_arm64_v8.0
```
When I copy in `native_musl_arm64.go` the corresponding lines of `native_musl_amd64.go`, it works:

```diff 
diff ../go/pkg/mod/kcl-lang.io/lib@v0.12.3/go/native/native_musl_arm64.go.orig ../go/pkg/mod/kcl-lang.io/lib@v0.12.3/go/native/native_musl_arm64.go
52c52,55
<                       return uintptr(C.kcl_service_call_with_length(cSvc, cMethod, cArgs, cArgsLen, cOutSize))
---
> 
>                       var cResultPtr *C.uint8_t
>                       cResultPtr = C.kcl_service_call_with_length(cSvc, cMethod, cArgs, cArgsLen, cOutSize)
>                       return uintptr(unsafe.Pointer(cResultPtr))
55c58,60
<                       C.kcl_free(C.uintptr_t(ptr), C.uint(len))
---
>                       cPtr := (*C.uint8_t)(unsafe.Pointer(ptr))
>                       cLen := C.uint32_t(len)
>                       C.kcl_free(cPtr, cLen)
```


### 2. What did you expect to see? (Required)

Compilation of the library.

### 3. What did you see instead (Required)

The errors above.

### 4. What is your KCL components version? (Required)

v0.12.3

--- a/go/native/native_musl_arm64.go
+++ b/go/native/native_musl_arm64.go
@@ -48,10 +48,15 @@ func initClient(pluginAgent uint64) {
 			defer C.free(unsafe.Pointer(cArgs))
 			cArgsLen := C.uint(argsLen)
 			cOutSize := (*C.uint)(unsafe.Pointer(outSize))
-			return uintptr(C.kcl_service_call_with_length(cSvc, cMethod, cArgs, cArgsLen, cOutSize))
+
+			var cResultPtr *C.uint8_t
+			cResultPtr = C.kcl_service_call_with_length(cSvc, cMethod, cArgs, cArgsLen, cOutSize)
+			return uintptr(unsafe.Pointer(cResultPtr))
 		}
 		free = func(ptr uintptr, len uint) {
-			C.kcl_free(C.uintptr_t(ptr), C.uint(len))
+			cPtr := (*C.uint8_t)(unsafe.Pointer(ptr))
+			cLen := C.uint32_t(len)
+			C.kcl_free(cPtr, cLen)
 		}
 		client = &NativeServiceClient{
 			svc: serviceNew(pluginAgent),

